websocket.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The go-ethereum Authors
```

Comment:
Logic taken directly from the go-ethereum repository - do not review

Reply:
Likewise here, why is this copied here instead of used from upstream?

Reply:
Unexported in upstream